repository_service_tuf_worker/repository.py

```diff
@@ -398,61 +417,128 @@ def bootstrap(
 
         return asdict(result)
 
-    def add_targets(self, payload: Dict[str, Any], update_state: str) -> None:
+    def publish_targets(self):
         """
-        Updates 'bins' roles metadata, assigning each passed target to the
-        correct bin.
+        Publish targets from SQL DB as persistent TUF Metadata in the backend
+        storage.
+        """
+
+        # lock to avoid race conditions
+        with self._redis.lock("publish_targets", timeout=5.0):
+            # get all delegated role names with unpublished targets
+            unpublished_roles = targets_crud.read_unpublished_rolenames(
+                self._db
+            )
 
-        Assignment is based on the hash prefix of the target file path. All
-        metadata is signed and persisted using the configured key and storage
-        services.
+            if len(unpublished_roles) == 0:
+                logging.debug("No new targets in delegated roles. Finishing")
+                return None
+
+            # initialize the new snapshot targets meta and published targets
+            # from DB SQL
+            new_snapshot_targets_meta: List[Tuple(str, int)] = []
+            db_published_targets: List[targets_models.RSTUFTargets] = []
+            for _, rolename in unpublished_roles:
+                # get the unpublished targets for the delegated, it will be use
+                # to update the database when Snapshot and Timestamp is
+                # published by `_update_timestamp`
+                db_targets = targets_crud.read_unpublished_by_rolename(
+                    db=self._db, rolename=rolename
+                )
+                logging.debug(f"{rolename}: New targets #: {len(db_targets)}")
+                db_published_targets += db_targets
+
+                # load the delegated targets role, clean the targets and add
+                # a new meta from the SQL DB.
+                # note: it might include targets from another parent task, it
+                # will speed the process of publishing new targets.
+                role = self._load(rolename)
+                role.signed.targets.clear()
+                role.signed.targets = {
+                    target[0]: TargetFile.from_dict(target[1], target[0])
+                    for target in targets_crud.read_all_add_by_rolename(
+                        self._db, rolename
+                    )
+                }
+
+                # update expiry, bump version and persist to the storage
+                self._bump_expiry(role, BINS)
+                self._bump_version(role)
```

Comment:
Does that mean that the version is updated for each delegated role that has targets updated? For example, if we update several targets spread across 5 delegated roles, the version bump will be +5?

Reply:
No, the bump will always be by +1.
Not on line 441, we start the loop by rolename. On line 455 I load the metadata from the role, add all targets, and bump it to the current version +1